clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -79,7 +79,9 @@ invariant_checker_t::report_if_false(per_shard_t *shard, bool condition,
     if (!condition) {
         std::cerr << "Trace invariant failure in T" << shard->tid_ << " at ref # "
                   << shard->stream->get_record_ordinal() << ": " << invariant_name
-                  << "\n";
+                  << "\nlast_timestamp: " << shard->last_timestamp_
+                  << "  instr_count_since_last_timestamp: "
```

Comment:
Better to work this into the existing message's location in a human-intended form, instead of printing variable names on a separate line.  Maybe "at ref # nnnn (mmm instrs since last timestamp kkkk): ...".